src/createAnimatedComponent.js

```diff
@@ -174,19 +180,23 @@ export default function createAnimatedComponent(Component) {
       }
     }
 
-    componentDidUpdate(prevProps) {
-      this._attachProps(this.props);
-      this._reattachNativeEvents(prevProps);
+    componentWillReceiveProps(newProps) {
+      this._attachProps(newProps);
+      if (this._component !== this._prevComponent) {
+        this._propsAnimated.setNativeView(this._component);
+      }
+      this._reattachNativeEvents(this.props);
+    }
+
+    _setComponentRef(c) {
+      this._prevComponent = this._component;
```

Comment:
avoid keeping ref to unmounted component. Maybe just have bool flag `refHasChanged`, seems to be all we need